api/src/wfl/source.clj

```diff
@@ -326,17 +326,18 @@
     (<= polling-interval minutes-since-poll)))
 
 (defn ^:private update-tdr-source
-  "Check for new data in TDR from `source`, create new snapshots,
+  "Check for new data in TDR from `Workload`'s `source`, create new snapshots,
   insert resulting job creation ids into database and update the
   timestamp for next time."
-  [{:keys [stopped] :as source}]
+  [{{:keys [stopped] :as source} :source :as workload}]
   (let [now (utc-now)]
     (when (and (not stopped) (tdr-source-should-poll? source now))
       (find-and-snapshot-new-rows source now)))
   (update-pending-snapshot-jobs source)
   ;; load and return the source table
-  (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
-    (load-tdr-source tx {:source_items (str (:id source))})))
+  (let [new-source (:source (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+                              (load-tdr-source tx {:source_items (str (:id source))})))]
+    (merge workload new-source)))
```

Comment:
I think the same advice applies here!

https://github.com/broadinstitute/wfl/pull/551/files#r777601418